### DISCUSSION TITLE
Need help compiling

### DISCUSSION THREAD

**Discussion by Shogun05** [QuEST-Kit/QuEST#394] [Q&A]
```
C:\Users\Shogun-Knight\Documents\EL\QuEST\QuEST\build>make
[ 80%] Built target QuEST
[ 90%] Linking C executable demo.exe
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: QuEST/libQuEST.a(QuEST_validation.c.obj):QuEST_validati:(.text+0x99b): undefined reference to `invalidQuESTInputError'
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: QuEST/libQuEST.a(QuEST_validation.c.obj):QuEST_validati:(.text+0x9af): undefined reference to `invalidQuESTInputError'
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: QuEST/libQuEST.a(QuEST_validation.c.obj):QuEST_validati:(.text+0xa3b): undefined reference to `invalidQuESTInputError'
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: QuEST/libQuEST.a(QuEST_validation.c.obj):QuEST_validati:(.text+0xa4f): undefined reference to `invalidQuESTInputError'
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: QuEST/libQuEST.a(QuEST_validation.c.obj):QuEST_validati:(.text+0xaa8): undefined reference to `invalidQuESTInputError'
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: QuEST/libQuEST.a(QuEST_validation.c.obj):QuEST_validati:(.text+0xb69): more undefined references to `invalidQuESTInputError' follow
collect2.exe: error: ld returned 1 exit status
make[2]: *** [demo.exe] Error 1
make[1]: *** [CMakeFiles/demo.dir/all] Error 2
make: *** [all] Error 2
```

Encountering this error whenever I try to make the project. The earlier command i ran was: 

`cmake .. -G "MinGW Makefiles" -D[VAR=VALUE]`

Where am i going wrong?

**Comment by TysonRayJones** [comment]
Hi there,
I hope you found a solution! I see you're compiling on Windows, and it looked like MinGW wasn't happy with the [weak symbol](https://github.com/QuEST-Kit/QuEST/blob/d4f75f724993b4af8e43a796e3c09ce24ae11670/QuEST/src/QuEST_validation.c#L235) for `invalidQuESTInputError`. Did you find out what caused this issue, or how to work around it?

**Reply by Shogun05** [reply]
Unfortunately I couldnt find a solution

**Reply by TysonRayJones** [reply]
Oh I'm sorry to hear that! Does closing the question mean you immediately gave up? :^) I'm happy to help you troubleshoot. Can you tell me your OS version, and the full commands with which you tried to compile through the terminal?

**Reply by Shogun05** [reply]
Haha yeah because i don't have much experience using make and cmake and it was kinda urgent to get the decoherence example compiled for our clg project. And as for the commands, the only one i remember running was the original command i gave in the issue. But gotta say cool project, I personally needed more experience with tools to actually run and build this :)

**Reply by TysonRayJones** [reply]
No problem at all! I totally get the need for getting things working quickly. It turns out anyway that MinGW (a "middleman" to get the GCC compiler working on Windows) doesn't even _support_ weak symbols as used by QuEST in `invalidQuESTInputError`) according to this [bug report](https://sourceware.org/bugzilla/show_bug.cgi?id=9687). So there's little I could have helped! 

This is useful to update QuEST's adverised compiler compatibility - although I _swear_ I once compiled QuEST with MinGW, hence why it's mentioned in the README (as a fallback when NMake fails). I'll look into it, thanks very much for the tip off!
